frontend/packages/component-library/src/tooltip/WithTooltip.tsx

```diff
@@ -23,143 +25,163 @@ const tailLengths = {
+    const tooltipRef = useRef<HTMLDivElement | null>(null);
+    const hideTimeoutRef = useRef<number | null>(null);
+    const suppressNextFocusRef = useRef(false);
+
+    const clearHideTimeout = () => {
+      if (hideTimeoutRef.current) {
+        clearTimeout(hideTimeoutRef.current);
+        hideTimeoutRef.current = null;
       }
     };
 
-    updateTooltipPositionIfShown();
-
-    window.addEventListener('resize', updateTooltipPositionIfShown);
-    return () => {
-      window.removeEventListener('resize', updateTooltipPositionIfShown);
-    };
-  }, [
-    showTooltip,
-    nodePosition,
-    tooltipProps.direction,
-    tailLength,
-    updateTooltipStyles,
-  ]);
-
-  // Effect to handle the Escape key to close the tooltip
-  useEffect(() => {
-    const handleKeyDown = (event: KeyboardEvent) => {
-      if (event.key === 'Escape') {
-        handleHideTooltip();
+    // Define the additional event handlers
+    const handleShowTooltip = (
+      show: boolean,
+      event: React.SyntheticEvent<HTMLElement>,
+      isTooltip: boolean = false,
+    ) => {
+      if (suppressNextFocusRef.current && event.type === 'focus') {
+        suppressNextFocusRef.current = false;
+        return;
+      }
+      setShowTooltip(show);
+      clearHideTimeout();
+      if (!isTooltip) {
+        setNodePosition(show ? (event.target as HTMLElement) : null);
       }
     };
-    if (showTooltip) {
-      document.addEventListener('keydown', handleKeyDown);
-    }
-    return () => {
-      document.removeEventListener('keydown', handleKeyDown);
+
+    const handleHideTooltip = () => {
+      hideTimeoutRef.current = window.setTimeout(() => {
+        setShowTooltip(false);
+        setNodePosition(null);
+      }, 100); // Allows for small but visible close delay
     };
-  }, [showTooltip]);
-
-  const tooltipStyleProps: React.CSSProperties = {
-    visibility: showTooltip ? 'visible' : 'hidden',
-    ...tooltipStyles,
-  };
-
-  // Check if children prop is a valid React element and clone it with ariaDescribedBy attribute
-  // and additional event handlers to make sure the tooltip is displayed correctly
-  const componentToWrap =
-    isValidElement<HTMLAttributes<HTMLElement>>(children) &&
-    cloneElement(children, {
-      'aria-describedby': tooltipProps.tooltipId,
-      onFocus: (event: React.FocusEvent<HTMLElement>) => {
-        handleShowTooltip(true, event);
-        children.props.onFocus?.(event);
-      },
-      onBlur: (event: React.FocusEvent<HTMLElement>) => {
-        handleHideTooltip();
-        children.props.onBlur?.(event);
-      },
-      onMouseEnter: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
-        handleShowTooltip(true, event);
-        children.props.onMouseEnter?.(event);
-      },
-      onMouseLeave: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
-        handleHideTooltip();
-        children.props.onMouseLeave?.(event);
+
+    useImperativeHandle(ref, () => ({
```

Comment:
I am reading that this is a built in react hook! I've never seen it/can't find an example in our codebase, so could you add a comment here to help folks (me) follow?

Reply:
Sure! just want to clarify - would you like me to add a comment in the code as well?

I did consider NOT using `useImperativeHandle` since it's not yet used in our codebase and instead, refactor `WithTooltip` so that tooltip visibility was moved to the parent component via props such as `showTooltip`, `onShowTooltip`, and `onHideTooltip`. But this would have been a bigger change and would require changes in all the other components that use `WithTooltip`.
Using `useImperativeHandle` allows us to expose `hideTooltip` to the parent with minimal change so that it can control when the tooltip is hidden. Hope this makes sense!

Reply:
Thank you for explaining! I think it could be worth adding something like what you just explained to me- esp the part about // useImperativeHandle allows us to manage show state without updating all tooltip parent components

Reply:
Sure - will do!